### ISSUE TITLE
Add comments/docs/specs to modules and functions

### ISSUE THREAD

**Issue by mbSmaga** [lambdaacademy/2016.04_erlang_xmpp_ofc#10]
In order to make the code more readable and easily understandable we should add comments/docs/specs, so the code will be easy to reason about and more friendly to newcomers.

**Comment by mentels** [comment]
I totally agree. I have some ideas on how to improve this. Will elaborate on that soon
